Lib/gzip.py

```diff
@@ -245,8 +245,13 @@ def __init__(self, filename=None, mode=None,
 
         self.fileobj = fileobj
 
-        if self.mode == WRITE:
-            self._write_gzip_header(compresslevel)
+        try:
+            if self.mode == WRITE:
+                self._write_gzip_header(compresslevel)
+        except:
+            # Avoid a ResourceWarning if the write fails, eg read-only file or KeyboardInterrupt
```

Comment:
Please truncate to 80 columns:

```suggestion
            # Avoid a ResourceWarning if the write fails,
            # eg read-only file or KeyboardInterrupt
```